### PR TITLE
DT-1296: ✨ Add in ktlint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,14 +1,4 @@
-[*.kt]
-indent_size = 2
-
-[*.kts]
-indent_size = 2
-
-[*.java]
```
Comment (petergphillips) [inline .editorconfig:7]: unnecessary for non java projects

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/TestFixtures.kt [hunk 2]
```diff
@@ -56,38 +57,41 @@ fun kotlinProjectDetails(projectDir: File) =
...
-          import org.springframework.boot.autoconfigure.SpringBootApplication
-          import org.springframework.boot.runApplication
-    
-          @SpringBootApplication
-          open class Application
-    
-          fun main(args: Array<String>) {
-            runApplication<Application>(*args)
-          }
+        package uk.gov.justice.digital.hmpps.app
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/TestFixtures.kt:60]: had to fix the formatting of the kotlin test files :grinning:

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/TestFixtures.kt [hunk 3]
```diff
@@ -56,38 +57,41 @@ fun kotlinProjectDetails(projectDir: File) =
...
+        }
+
       """.trimIndent(),
     buildScriptName = "build.gradle.kts",
     buildScript =
       """
           plugins {
             id("uk.gov.justice.hmpps.gradle-spring-boot") version "0.1"
           }
+
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/TestFixtures.kt:79]: this adds a newline to the end of the file (required for ktlint)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KtlintPluginManagerTest.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    val result = buildProjectAndFail(projectDir, "check")
+    assertThat(result.output)
+      .contains("Wildcard import (cannot be auto-corrected)")
+      .contains("Unexpected indentation (4) (should be 2)")
+  }
+
+  @ParameterizedTest
+  @MethodSource("defaultProjectDetails")
+  fun `The sonar properties file is copied into the project`(projectDetails: ProjectDetails) {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KtlintPluginManagerTest.kt:56]: Should we have a similar test for editorconfig in this test, and move this test to another test class?

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KtlintPluginManagerTest.kt:56]: copy and paste fail :facepalm:

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KtlintPluginManagerTest.kt:56]: already all exists in BaseConfigManagerGradleTest and tested sonar stuff fully there, just wanted to test editor config copying here

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**